### ISSUE TITLE
Stack block: can't set custom height on reusable blocks

### ISSUE THREAD

**Issue by motla** [WordPress/gutenberg#50734]
### Description

Let's say I have a stack block, containing paragraph blocks and a reusable block. We apply a minimum height to this stack block (here 260px):
<img width="983" alt="image" src="https://github.com/WordPress/gutenberg/assets/16179413/82a90ef7-18c2-448d-9a23-3925b17eac88">

Now let's say we want the reusable block to grow to the maximum height available. We can do that by setting the `Block‑>Styles‑>Dimensions‑>Height` property to `Fill`:
<img width="984" alt="image" src="https://github.com/WordPress/gutenberg/assets/16179413/e72d5199-180c-410f-ad34-96c944a809b3">

**However, when we save the page, this property is lost.**

---

We can see that in the code editor:
```html
<!-- wp:block {"ref":478} /-->
```
should have become:
```html
<!-- wp:block {"ref":478,"style":{"layout":{"selfStretch":"fill","flexSize":null}}} /-->
```
like if it would be applied on a regular paragraph, **but it is not**.

---

### Step-by-step reproduction instructions

Paste this code in a new page:

```html
<!-- wp:group {"style":{"dimensions":{"minHeight":"260px"}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
<div class="wp-block-group" style="min-height:260px">
<!-- wp:paragraph -->
<p>My header paragraph</p>
<!-- /wp:paragraph -->

<!-- wp:block {"ref":478} /-->

<!-- wp:paragraph -->
<p>My footer paragraph</p>
<!-- /wp:paragraph -->
</div>
<!-- /wp:group -->
```

Try to set `Block‑>Styles‑>Dimensions‑>Height` property to `Fill` on the reusable block, then save the page. This property is lost.

---

### Environment info

- Wordpress 6.2.1, Twenty Twenty-Three theme Version: 1.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by miminari** [comment]
@motla Hi!
I attempted to reproduce the test.
However, I was unable to fully replicate the issue on WordPress 6.4.2.
This is because the Pattern does not seem to have the Dimensions functions.
I'm unsure if this is a specification or a bug, but it would be beneficial if inner blocks could be made to have the Dimensions functions, even when using reused blocks.

I have attached a video of a similar situation created with 6.4.2 for reference.
https://github.com/WordPress/gutenberg/assets/487207/6475d950-ec8d-4a6e-906e-7c34d7e82237
